src/IO/WriteBuffer.h

```diff
@@ -109,6 +109,8 @@ class WriteBuffer : public BufferBase
         next();
     }
 
+    virtual void preFinalize() {}
```

Comment:
No comments - inacceptable.

Reply:
It's just a draft yet :)

Reply:
The logic is the same as `prefetch` in `ReadBuffer`.
I don't like the name. It should be a point when all data is written, and we can send the rest of the data to s3